### PR TITLE
Install SNAPSHOTs to mavenLocal() instead of ANDROID_HOME

### PR DESCRIPTION
**Pull request by cmelchior**
This PR install the java library + gradle plugin in mavenLocal() instead of ANDROID_HOME.
For some reason `uploadArchives` doesn't work for me, while `install` does. Which is a bit odd since @emanuelez said the opposite was true for him.

I changed the caching strategy to never cache snapshots for examples which makes it easier to update them during development.  

I also added an extra task for deploying both plugin + library `installRealmJava`.

@realm/java 
@emanuelez

### CODE REVIEW

**Comment by emanuelez** [comment]
At a first quick glance if this works you can remove the maven-publish plugin.

**Comment by cmelchior** [comment]
Isn't the `maven-publish` plugin responsible for handling the `publishing` block in the `realm` folder?

**Comment by emanuelez** [comment]
You are absolutely right.
I would really love to deploy to a folder in the repo. That would make sure that a git clean will really reset the environment

**Comment by cmelchior** [comment]
There already is a `cleanLocalMavenRepos` task that does exactly that?

**Comment by emanuelez** [comment]
Yes, but that's a horrible hack IMO

**Comment by cmelchior** [comment]
Could you use a `post-checkout` git hook and call that method? I doesn't look like there is a clean hook?

**Comment by emanuelez** [comment]
Adding more horrible hacks? :)

**Comment by cmelchior** [comment]
You're right :) In any case that is something that is not done in the current solution, so shouldn't really impact this PR?

File: realm/gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -4,3 +4,4 @@ distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
 distributionUrl=https\://services.gradle.org/distributions/gradle-2.7-bin.zip
+set JAVA_OPTS="-Xmx512m"
```
Comment (zaki50) [inline realm/gradle/wrapper/gradle-wrapper.properties]: In #1707 we use gradle.properties.

Should this vm args also go to that file?

Comment (cmelchior) [inline realm/gradle/wrapper/gradle-wrapper.properties]: I must confess I am not entirely sure about the difference (if any) between the two files, but if it doesn't make difference, then yes, it should probably be moved to the `gradle.properties` file instead

Comment (zaki50) [inline realm/gradle/wrapper/gradle-wrapper.properties]: This file only affects gradle wraper(gradlew).

`gradle.properties` affects any gradle execution which is not limited to gradle wrapper.

And I realize that `set FOO` is not a syntax for properties file.
I suspect that this line has no effect at all.

Comment (cmelchior) [inline realm/gradle/wrapper/gradle-wrapper.properties]: I'll remove it again then. I was trying to fix a Java Heap space bug when compiling

Comment (nhachicha) [inline realm/gradle/wrapper/gradle-wrapper.properties]: have you tried the following inside `gradle.properties`
```groovy
org.gradle.jvmargs=-Xms256m -Xmx1024m
```
more here https://docs.gradle.org/current/userguide/build_environment.html

File: build.gradle [hunk 2]
```diff
@@ -75,17 +75,24 @@ task connectedCheck(type:GradleBuild) {
 }
 
 task installGradlePlugin(type:GradleBuild) {
-    description = 'Install the Realm Gradle plugin to the Maven Repo in ANDROID_HOME'
+    description = 'Install the Realm Gradle plugin to mavenLocal()'
```
Comment (nhachicha) [inline build.gradle]: to-> into

File: build.gradle [hunk 3]
```diff
@@ -75,17 +75,24 @@ task connectedCheck(type:GradleBuild) {
...
     group = 'Install'
     buildFile = file('gradle-plugin/build.gradle')
-    tasks = ['uploadArchives']
+    tasks = ['install']
 }
 
 task installRealm(type:GradleBuild) {
     group = 'Install'
-    description = 'Install the artifacts of Realm project to the Maven Repo in ANDROID_HOME'
+    description = 'Install the artifacts of Realm project to mavenLocal()'
```
Comment (nhachicha) [inline build.gradle]: to-> into

File: build.gradle [hunk 4]
```diff
@@ -75,17 +75,24 @@ task connectedCheck(type:GradleBuild) {
...
     buildFile = file('realm/build.gradle')
-    tasks = ['uploadArchives']
+    tasks = ['install']
+}
+
+task installRealmJava(type:Task) {
+    dependsOn installGradlePlugin
+    dependsOn installRealm
+    group = 'Install'
+    description = 'Install the Realm library and Gradle plugin to mavenLocal()'
```
Comment (nhachicha) [inline build.gradle]: to->into

File: realm/gradle/wrapper/gradle-wrapper.properties (hunk 1 above)
Comment (cmelchior) [inline realm/gradle/wrapper/gradle-wrapper.properties]: I just got another Java Heap space error. I'll try to add the above to `gradle.properties`

**Comment by cmelchior** [comment]
Updated with comments.

**Comment by zaki50** [comment]
:+1:

**Comment by cmelchior** [comment]
Bump @realm/java

**Comment by kneth** [comment]
:+1:

**Pull request merged**